src/utils/std.ml

```diff
@@ -346,6 +346,14 @@ module Result = struct
   type ('a, 'e) t = ('a, 'e) result =
   | Ok of 'a
   | Error of 'e
+
+  let map ~f = function
+  | Error err -> Error err
+  | Ok x -> Ok (f x)
+
+  let bind ~f = function
+  | Error err -> Error err
+  | Ok x -> f x
```

Comment:
Those are already part of the stdlib (minus the labels).
If we want labelled versions we can simply reuse the official implementations:

```
let map ~f r = Result.map f r
```